text/0000-components.md

```diff
@@ -0,0 +1,261 @@
+those which can be solved with a distributed Byzantine-fault-tolerant database.
+For example, while much of the Grid daemon as described below is related to the
+implementation of Sawtooth functionality (such as state delta export), we
+anticipate the Grid Daemon will contain any Grid code which runs as a service.
+
+Grid contains the following components:
+
+- A daemon process called “gridd” (pronounced grid-dee) or Grid daemon
+- A command line tool called “grid” or Grid CLI
+- A Grid SDK
+- Smart contracts
+- Documentation
+
+![Grid Components Overview](../images/grid_components_overview.svg)
+
+## Daemon (gridd)
+
+Hyperledger Grid provides several services which are always running. The
+services are of three types: 1) application-facing services; 2) integration
+services; and 3) network-facing services. An example of a application-facing
+service is the REST API which is the primary point of interaction between
+applications and the Grid daemon. An example of an integration service is state
+delta export, which integrates with a Sawtooth validator to materialize
+Sawtooth’s global state into a PostgreSQL database that is used within the Grid
+daemon. Network-facing services interact with other Grid daemons to form a Grid
+network and are not directly covered here; however, if the Sawtooth validator
+was 100% Rust code, conceptually it could be compiled into the Grid daemon and
+be considered a network-facing service.
+
+The Grid daemon process contains all of the components which run as
+a persistent process as part of Hyperledger Grid. An intentional part of the
+Grid architecture is to minimize persistent processes to a bare minimum with
+the ideal state being a single Grid daemon process. This is architecturally
+distinct from Hyperledger Sawtooth’s design which favors flexibility by
+utilizing numerous processes which connect to each other. The impact of the
+Grid architecture is that it will be simple to deploy a Grid daemon, but
+suggests the project should use a single programming language (Rust) for all
+persistent process code.
+
+Initially, the Grid daemon contains:
+
+- A REST API
+- State Delta Export
+
+Additional components will be added to the Grid daemon in the future.
+
+### REST API
+
+The Grid daemon’s REST API is the primary interface between applications and
+Hyperledger Grid.
+
+The Grid REST API provides an interface for applications to:
+
+- Submit transactions
+- Read/query schema data
+- Read/query Pike identity data
+- Read/query product data
+- Read/query track and trace data
+
+The REST API’s functionality will continue to expand to match the capabilities
+provided by Hyperledger Grid. This includes additional APIs for data-related
+read/query, Grid management and configuration (potentially including management
+and configuration of the Grid daemon itself), application key “escrow”, and any
+other functionality which is provided by Grid for applications.
```

Comment:
For clarity, are `write` methods not going to be exposed through the Grid daemon ReST API (outside of management opportunities called out).

For example, the ability to modify `schema data` is a separate concern and bounded context.

Asking as perhaps that should be declared here with pointers to where these types of activities will live.